fenix/app/src/main/java/org/mozilla/fenix/tabhistory/TabHistoryController.kt

```diff
@@ -4,22 +4,18 @@
 
 package org.mozilla.fenix.tabhistory
 
-import androidx.navigation.NavController
 import mozilla.components.feature.session.SessionUseCases
 
 interface TabHistoryController {
     fun handleGoToHistoryItem(item: TabHistoryItem)
 }
 
 class DefaultTabHistoryController(
-    private val navController: NavController,
     private val goToHistoryIndexUseCase: SessionUseCases.GoToHistoryIndexUseCase,
     private val customTabId: String? = null,
 ) : TabHistoryController {
 
     override fun handleGoToHistoryItem(item: TabHistoryItem) {
-        navController.navigateUp()
```

Comment:
@iorgamgabriel @mcarare I find this suspicious. Have we tried simply changing this to `popBackStack`. 

Alternatively, have we tried passing a lamda similar that sets bottom sheet behavior to state hidden. Similar to examples @gabrielluong shared. 

From the docs

_Attempts to navigate up in the navigation hierarchy. Suitable for when the user presses the "Up" button marked with a left (or start)-facing arrow in the upper left (or starting) corner of the app UI.
The intended behavior of Up differs from Back when the user did not reach the current destination from the application's own task. e.g. if the user is viewing a document or link in the current app in an activity hosted on another app's task where the user clicked the link. In this case the current activity (determined by the context used to create this NavController) will be finished and the user will be taken to an appropriate destination in this app on its own task.
Returns:
true if navigation was successful, false otherwise_

Reply:
@rahulsainani You are welcome to try other paths to fix this. As long we don't have a STR I am not sure we can do more than a speculative fix.